Headlines/src/Categories/UIImageEffects.swift

```diff
@@ -0,0 +1,281 @@
+        }
+        
+        let FLTEPSILON = CGFloat(Float.ulpOfOne)
+        let screenScale = UIScreen.main.scale
+        let imageRect = CGRect(origin: CGPoint.zero, size: size)
+        var effectImage = self
+        
+        let hasBlur = blurRadius > FLTEPSILON
+        let hasSaturationChange = fabs(saturationDeltaFactor - 1.0) > FLTEPSILON
+        
+        if hasBlur || hasSaturationChange {
+            func createEffectBuffer(_ context: CGContext) -> vImage_Buffer {
+                let data = context.data
+                let width = vImagePixelCount(context.width)
+                let height = vImagePixelCount(context.height)
+                let rowBytes = context.bytesPerRow
+                
+                return vImage_Buffer(data: data, height: height, width: width, rowBytes: rowBytes)
+            }
+            
+            UIGraphicsBeginImageContextWithOptions(size, false, screenScale)
+            guard let effectInContext = UIGraphicsGetCurrentContext() else { return  nil }
+            
+            effectInContext.scaleBy(x: 1.0, y: -1.0)
+            effectInContext.translateBy(x: 0, y: -size.height)
+            effectInContext.draw(cgImage, in: imageRect)
+            
+            var effectInBuffer = createEffectBuffer(effectInContext)
+            
+            
+            UIGraphicsBeginImageContextWithOptions(size, false, screenScale)
+            
+            guard let effectOutContext = UIGraphicsGetCurrentContext() else { return  nil }
+            var effectOutBuffer = createEffectBuffer(effectOutContext)
+            
+            
+            if hasBlur {
+                // A description of how to compute the box kernel width from the Gaussian
+                // radius (aka standard deviation) appears in the SVG spec:
+                // http://www.w3.org/TR/SVG/filters.html#feGaussianBlurElement
+                //
+                // For larger values of 's' (s >= 2.0), an approximation can be used: Three
+                // successive box-blurs build a piece-wise quadratic convolution kernel, which
+                // approximates the Gaussian kernel to within roughly 3%.
+                //
+                // let d = floor(s * 3*sqrt(2*pi)/4 + 0.5)
+                //
+                // ... if d is odd, use three box-blurs of size 'd', centered on the output pixel.
+                //
+                let inputRadius = blurRadius * screenScale
+                let d = floor(inputRadius * 3.0 * CGFloat(sqrt(2 * .pi) / 4 + 0.5))
+                var radius = UInt32(d)
+                if radius % 2 != 1 {
+                    radius += 1 // force radius to be odd so that the three box-blur methodology works.
+                }
+                
+                let imageEdgeExtendFlags = vImage_Flags(kvImageEdgeExtend)
+                
+                vImageBoxConvolve_ARGB8888(&effectInBuffer, &effectOutBuffer, nil, 0, 0, radius, radius, nil, imageEdgeExtendFlags)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/b49c86a9b05bec3bbe17833b5a716a8293c42cf1/Headlines/src/Categories/UIImageEffects.swift#L201"></span>Line should be 120 characters or less: currently 131 characters</td>
    </tr>
  </tbody>
</table>